hngprojects/hng_boilerplate_nestjs
# feat: create new comment

**What does the PR do?**
This PR introduces new API endpoints to create comments. It includes server-side logic for handling comment creation, along with appropriate validation and error handling.

How should this be manually tested?

Create Comment:

Send a POST request to /api/v1/comments/add with a valid CreateCommentDto payload and a valid user ID in the request.
Example:

`POST /api/v1/comments/add`

{
  "model_id": "1234",
  "model_type": "post",
  "content": "This is a comment"
}

Successful Response

On a successful comment creation, the API should return a 201 SUCCESS status code.

```json
{
      "status": "status_code",
      "message": "Comment created successfully",
      "data": {
            "model_id": "1234",
            "model_type": "John Doe",
            "content":"Comment content as supplied by the user",
            }
      "commentedBy" : "John Smith"
}
```

Error Handling:

When the current user does not have permission or is not found:

```json
{
          "status": "error",
          "message": "User nof found",
          "status_code": "404",
 }
```

Error During coment creation

If the comment creation is not successful:

```json
{
  "status": "Error",
  "message": "Error creating comment",
  "status_code": 500
}
```

 **Checklist of what has been done**
- [X ] Define API endpoint to create comment.
- [X ]  Validate and handle authentication token.
- [ X]  Implement logic to create a comment.
- [ X]  Handle success and error responses.
- [ X]  Test Successful response

**API Test Screenshot**
![Screenshot from 2024-08-07 15-42-42](https://github.com/user-attachments/assets/5795b8d3-fa59-45c5-af15-7f41aafca035)

** Jest Test Screenshot **
![Screenshot from 2024-08-07 15-46-24](https://github.com/user-attachments/assets/19fbf8e7-bc34-474c-b6c1-2623553258d3)


Reference Issue: [Link to Issue](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/151)
